src/components/LabelWarning.vue

```diff
@@ -0,0 +1,175 @@
+    },
+    flowGroup: {
+      type: Object,
+      default: () => {}
+    },
+    iconSize: {
+      type: String,
+      required: false,
+      default: null
+    },
+    always: {
+      type: Boolean,
+      default: true
+    },
+    location: {
+      type: String,
+      default: ''
+    }
+  },
+  data() {
+    return {
+      infoMessage: '',
+      docsName: '',
+      docsLink: ''
+    }
+  },
+  computed: {
+    ...mapGetters('tenant', ['tenant']),
+    ...mapGetters('agent', ['agents']),
+    agentOrLabel() {
+      if (!this.agents || !this.agents.length) return 'Agent Problem'
+      if (!this.labelsAlign) return 'Label Problem'
+      return 'Flow and Agent Labels'
+    },
+    agentLabels() {
+      if (this.agents) {
+        const labels = this.agents.reduce((accumulator, agent) => {
+          accumulator.push(agent.labels)
+          return accumulator
+        }, [])
+        return labels
+      }
+      return null
+    },
+    flowLabels() {
+      const labels = this.flowGroup?.labels || this.flow?.environment?.labels
+      return labels
+    },
+    labelsAlign() {
+      if (!this.agents) return
+      if (!this.agents.length) {
+        this.labelMessage(
+          'You have no live agents.  Flow runs will not be picked up.',
+          'agents',
+          'https://docs.prefect.io/orchestration/agents/overview.html'
+        )
+        return false
+      }
+      if (
+        !this.flowLabels.length &&
+        this.agentLabels.every(arrayOfLabels => arrayOfLabels.length > 0)
+      ) {
+        this.labelMessage(
+          'To let the agent pick up this flow run, you may need to add labels to your flow.',
+          'Labels and Flow Affinity',
+          'https://docs.prefect.io/orchestration/agents/overview.html#flow-affinity-labels'
+        )
+        return false
+      } else {
+        let matchingLabels = 0
+        if (this.agentLabels) {
+          this.agentLabels.forEach(array => {
+            if (this.flowLabels.every(label => array.includes(label)))
+              matchingLabels++
+          })
+        }
+        if (matchingLabels > 0) {
+          this.labelMessage(
+            'Your flow and agent labels look good.',
+            'Labels and Flow Affinity',
+            'https://docs.prefect.io/orchestration/agents/overview.html#flow-affinity-labels'
+          )
+          return true
+        } else {
+          this.labelMessage(
+            'It looks like you have a label on your flow that does not match your agent labels. To let the agent pick up this flow run, you need to align your flow and agents label.',
+            'Labels and Flow Affinity',
+            'https://docs.prefect.io/orchestration/agents/overview.html#flow-affinity-labels'
+          )
+          return false
+        }
+      }
+    }
+  },
+  methods: {
+    labelMessage(message, docsName, link) {
+      this.infoMessage = message
+      this.docsName = docsName
+      this.docsLink = link
+    }
```

Comment:
I think we could increase the clarity of this component and the clarity and idempotence of the `labelsAlign` prop by moving each of the props you set here to their own computed props whose values are determined by the value of `labelsAlign`. Something like this:

```js
labelsAlign() {
  return !this.flowLabels.length && this.agentLabels.every(labels => !!labels?.length)
},
docsLink() {
  return this.agents?.length ? 'https://docs.prefect.io/orchestration/agents/overview.html#flow-affinity-labels' : 'https://docs.prefect.io/orchestration/agents/overview.html'
}
```
etc. etc.